aten/src/ATen/native/xnnpack/Common.h

```diff
@@ -0,0 +1,81 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#ifdef USE_XNNPACK
+
+#include <xnnpack.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+struct Layout final {
+  // 4D Activation Maps
+  struct Activation4D final {
+    static constexpr size_t Batch = 0u;
+    static constexpr size_t Channels = 1u;
+    static constexpr size_t Height = 2u;
+    static constexpr size_t Width = 3u;
+  };
+
+  // ND Activation Maps
+  struct ActivationND final {
+    // Some operators may not be limited to 4 dimensional tensors. In that scenario,
+    // XNNPACK denotes that operator with an _nc suffix and expects all dimensions,
+    // except channels, to be flattened into one argument: batch_size.
+    static int64_t Batch(const IntArrayRef tensor) {
+      if (C10_UNLIKELY(tensor.empty())) {
+        return -1;
```

Comment:
Can you take a follow-up task to make sure the XNNPACK ops work with batch size 0 (and add a unit test for this)?

Reply:
Is batch size zero valid?  Isn't the minimum batch size one?

Reply:
Yes, batch size zero is valid.

Reply:
And recently we made sure that common ops work with batch size 0, because this is a common request #12013. Unfortunately, xnnpack integration does not go through common tests for nn.Conv2d, or you'd see breakages.

Reply:
XNNPACK skips computations for operators with batch size 0, with both `xnn_setup_*` and `xnn_run_operator` returning `xnn_status_success`.